pkg/server/plugin/zmq/rpc.go

```diff
@@ -176,6 +194,54 @@ func (p *zmqTransport) ipc(req *pluginrequest.Request) (out []byte, err error) {
 	return
 }
 
+func (p *zmqTransport) handleRequest(parcel *Parcel) {
+	workerID := parcel.Worker
+	requestID := parcel.RequestID
+	bounceCount := parcel.BounceCount
+	buffer := parcel.Frame
+	responseChan := parcel.RespChan
+
+	reader := bytes.NewReader(buffer)
+	data := map[string]interface{}{}
+	if jsonErr := json.NewDecoder(reader).Decode(&data); jsonErr != nil && jsonErr != io.EOF {
+		// TODO(b123400): handle json error
+		return
+	}
+
+	payloadData := data["payload"].(map[string]interface{})
+	method := data["method"].(string)
+	path := data["path"].(string)
+
+	payload := &router.Payload{
+		Context: context.Background(),
+		Meta: map[string]interface{}{},
+		Data: payloadData,
+		AccessKey: router.MasterAccessKey,
+	}
+
+	responseWriter := &zmqResponseWriter{}
+	resp := router.NewResponse(responseWriter)
+
+	newCtx := context.WithValue(payload.Context, ZMQWorkerIDContextKey, workerID)
+	newCtx = context.WithValue(newCtx, ZMQRequestIDContextKey, requestID)
+	newCtx = context.WithValue(newCtx, ZMQBounceCountContextKey, bounceCount)
+	payload.Context = newCtx
+
+	// TODO(b123400), put real path and method here
+	p.router.HandlePayload(path, method, payload, resp)
+
+	responseChan <- responseWriter.response
+}
+
+func (p *zmqTransport) listenRequests() {
+	for {
+		select {
+		case parcel := <- p.broker.ReqChan:
+			go p.handleRequest(parcel)
```

Comment:
I meant instead of having people write things like this:

```go
parcel = makeParcel()
broker.ReqChan <- parcel
// do other things
```

Let them write this:

```go
go func() {
	parcel = makeParcel()
	broker.ReqChan <- parcel
}()

// do other things immediately
```

If you don't want to block, demand it explicitly. The problem of this `listenRequests` is that it does far too little things to justify itself a thread of execution. Consider it in Erlang or Elixir, would you spawn a process just to pass massage hand to hand without doing anything? I bet no.

In this request-response case, it might even be better to make a function to hide the complexity of asynchrony:

```go
// purely imaginary
func doRequest(parcel Parcel) ([]byte, error) {
	respChan = make(chan []byte)
	errChan = make(chan error)

	parcel.respChan = respChan
	parcel.errChan = errChan

	broker.ReqChan <- parcel
	select {
		resp := <- respChan:
			return resp, nil
		err := <- errChan:
			return nil, err
		// other things here, like timeout
	}
}

// some busy code here...
go func() {
	parcel = make_parcel()
	resp, err = doRequest(parcel)
	// now process
}()

// do other things immediately
```